src/promptflow/promptflow/_core/tool_meta_generator.py

```diff
@@ -119,7 +125,12 @@ def _parse_tool_from_function(f):
     try:
         inputs, _, _ = function_to_interface(f)
     except Exception as e:
-        raise BadFunctionInterface(f"Failed to parse interface for tool {f.__name__}, reason: {e}") from e
+        error_type_and_message = f"({e.__class__.__name__}) {e}"
+        raise BadFunctionInterface(
+            message_format="Parse interface for tool {tool_name} failed: {error_type_and_message}",
```

Comment:
Add quota to tool_name:

`{tool_name}` -> `'{tool_name}'`

Reply:
Done.